halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/Main.java

```diff
@@ -10,15 +10,16 @@
 public class Main {
   public static void main(String[] args) {
     GlobalOptions globalOptions = GlobalOptions.getGlobalOptions();
-    JCommander jc = new JCommander(globalOptions);
 
-    HalCommand hal = new HalCommand(jc);
+    HalCommand hal = new HalCommand();
+    JCommander jc = new JCommander(hal);
+    hal.setCommander(jc).configureSubcommands();
 
     try {
       jc.parse(args);
     } catch (ParameterException e) {
       System.out.println(e.getMessage());
-      jc.usage();
+      return;
```

Comment:
Good point, let me see if there is a nice way to do that in java without providing a stack trace (forwarding the exception)